libvast/src/system/accountant.cpp

```diff
@@ -37,8 +40,75 @@
 #include <limits>
 
 namespace vast {
+
 namespace system {
 
+template <class V>
+bool assign(V& v, const caf::settings& m, std::string_view key) {
+  auto it = m.find(key);
+  if (it == m.end())
+    return true;
+  if constexpr (caf::detail::tl_contains<caf::config_value::variant_type::types,
+                                         V>::value) {
+    auto x = caf::get_if<V>(&it->second);
+    if (!x)
+      return false;
+    v = *x;
+    return true;
+  } else {
+    auto x = caf::get_if<std::string>(&it->second);
+    if (!x)
+      return false;
+    auto f = x->begin();
+    return parse(f, x->end(), v);
+  }
+}
+
+template <class V>
+bool assign(V& v, const map& m, const data& key) {
+  auto it = m.find(key);
+  if (it == m.end())
+    return true;
+  if constexpr (caf::detail::tl_contains<data::types, V>::value) {
+    auto x = caf::get_if<V>(&it->second);
+    if (!x)
+      return false;
+    v = *x;
+    return true;
+  } else {
+    auto x = caf::get_if<std::string>(&it->second);
+    if (!x)
+      return false;
+    auto f = x->begin();
+    return parse(f, x->end(), v);
+  }
+}
```

Comment:
I really don't like these being named just `assign`. Can you find a better name and add a comment explaining what these functions do? It was not immediately clear from reading the code.

Reply:
I've since renamed the function to `absorb`, but @dominiklohmann is also not happy with that and suggests `copy_from_settings` which I find to clunky. Maybe someone else from @tenzir/backend can be the tie-breaker here.